contracts/DCAHub/DCAHubPositionHandler.sol

```diff
@@ -196,7 +196,10 @@ abstract contract DCAHubPositionHandler is ReentrancyGuard, DCAHubConfigHandler,
       _increase ? IDCAPermissionManager.Permission.INCREASE : IDCAPermissionManager.Permission.REDUCE
     );
 
-    if (_increase) _assertTokensAreAllowed(_userPosition.from, _userPosition.to);
+    if (_increase) {
+      _assertTokensAreAllowed(_userPosition.from, _userPosition.to);
+      _updateActiveIntervalsAndOracle(_userPosition.from, _userPosition.to, _userPosition.swapIntervalMask);
```

Comment:
Let's check, but this should be cheaper in terms of gas. Also, not direct replacement, but I'm on the cellphone 

```suggestion
     (address _tokenA, address _tokenB) = TokenSorting.sortTokens(_from, _to);
     activeSwapIntervals[_tokenA][_tokenB] |= _mask;   
```

Reply:
I'm on desktop, so I can write it 😆 
```
(address _tokenA, address _tokenB) = TokenSorting.sortTokens(_userPosition.from, _userPosition.to);
activeSwapIntervals[_tokenA][_tokenB] |= _userPosition.swapIntervalMask;  
```